mapping/index.go

```diff
@@ -447,6 +448,33 @@ func (im *IndexMappingImpl) FieldAnalyzer(field string) string {
 	return im.AnalyzerNameForPath(field)
 }
 
+// FieldMappingForPath returns the mapping for a specific field 'path'.
+func (im *IndexMappingImpl) FieldMappingForPath(path string) FieldMapping {
+	if im.TypeMapping != nil {
+		for _, v := range im.TypeMapping {
+			for field, property := range v.Properties {
+				for _, v1 := range property.Fields {
+					if field == path {
+						// Return field mapping if the name matches the path param.
+						return *v1
+					}
+				}
+			}
+		}
+	}
+
+	for field, property := range im.DefaultMapping.Properties {
+		for _, v1 := range property.Fields {
+			if field == path {
+				// Return field mapping if the name matches the path param.
+				return *v1
+			}
+		}
+	}
+
+	return FieldMapping{}
```

Comment:
i'm thinking if its better we return a nil over here, avoiding allocation of a struct over here, and in the caller if the field mapping is nil, let the caller handle accordingly. for example, in the searcher construction for knn, if the field mapping is nil, we use the default SimilarityDefaultVal. any thoughts on this?

Reply:
I don't think I can do that since I'm not returning a pointer over here.
I will add a check in the caller though to use the default val if it's "".

Reply:
Pls resolve this if it makes sense, thanks!